src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseService.java

```diff
@@ -59,6 +60,22 @@ public CaseDetails createNewCaseDetails(String caseTypeId, String jurisdictionId
         return caseDetails;
     }
 
+    /**
+     * @param content Data received from the client.
+     * @param jurisdictionId of the case.
+     * @return <code>Optional&lt;CaseDetails&gt;<code/> - CaseDetails wrapped in Optional
+     */
+    public Optional<CaseDetails> populateCurrentCaseDetailsWithUserInputs(CaseDataContent content, String jurisdictionId) {
+        try {
+            CaseDetails caseDetails = getCaseDetails(jurisdictionId, content.getCaseReference());
+            content.getEventData().forEach((key, value) -> caseDetails.getData().put(key, value));
+
+            return Optional.of(caseDetails);
```

Comment:
you should always use Optional.ofNullable rather than Optional.of. Much safer. If someone changes `getCaseDetails` to return null, your code will break

Reply:
Beautiful and very valuable